### DISCUSSION TITLE
Sort gallery by ACF

### DISCUSSION THREAD

**Discussion by marmulaq** [elementor/elementor#24382] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Hello,
I have an ACF field of type: text, default value: "undefined".
This field is displayed in the Media gallery for images where I assign a value to them (eg "Nature").
In the Element editor's Filterable gallery, I'll set Dynamic Links and select ACF field from the list.
The gallery works, the images that I mark in the ACF field as "Nature" are filtered by the ACF field "Nature", but the text listed in the ACF field ("Nature") is not displayed in the header of the gallery with the filter. Only the default field name is displayed ("undefined" in my case).
When I leave the "default value:" field in the ACF definition blank, no text appears in the gallery header (I would expect "Nature").
However, clicking on an empty space with no text in the gallery header filters out images marked with the "Nature" field correctly.
So the filter works correctly, but the text that is written in the corresponding ACF field is not displayed.
Thank you for the solution.
Sincerely, M.G.

![Výstřižek_01](https://github.com/elementor/elementor/assets/111281435/6ef5a2aa-8e32-4a43-be72-552d410c088a)
![Výstřižek_02](https://github.com/elementor/elementor/assets/111281435/8c22c8a0-d83a-46bb-8013-c8530ce4f83c)
![Výstřižek_03](https://github.com/elementor/elementor/assets/111281435/702b7b6e-9a7a-4f8b-a12d-41de27b6ab71)
![Výstřižek_04](https://github.com/elementor/elementor/assets/111281435/8734c941-1592-4e3d-90d6-397fdf7cb503)

### Describe the solution you'd like

Hello,
I have an ACF field of type: text, default value: "undefined".
This field is displayed in the Media gallery for images where I assign a value to them (eg "Nature").
In the Element editor's Filterable gallery, I'll set Dynamic Links and select ACF field from the list.
The gallery works, the images that I mark in the ACF field as "Nature" are filtered by the ACF field "Nature", but the text listed in the ACF field ("Nature") is not displayed in the header of the gallery with the filter. Only the default field name is displayed ("undefined" in my case).
When I leave the "default value:" field in the ACF definition blank, no text appears in the gallery header (I would expect "Nature").
However, clicking on an empty space with no text in the gallery header filters out images marked with the "Nature" field correctly.
So the filter works correctly, but the text that is written in the corresponding ACF field is not displayed.
Thank you for the solution.
Sincerely, M.G.
![Výstřižek_01](https://github.com/elementor/elementor/assets/111281435/b1124b5c-d350-49ab-a32a-083f12dfdaf3)
![Výstřižek_02](https://github.com/elementor/elementor/assets/111281435/de82d8a1-b778-4999-b8e9-d9080831858b)
![Výstřižek_03](https://github.com/elementor/elementor/assets/111281435/52c857ec-700d-4e89-8a0d-3b51554bef35)
![Výstřižek_04](https://github.com/elementor/elementor/assets/111281435/b5bfce4f-ab1f-4cb2-af0a-5ae83c46609c)

**Comment by theaminulai** [comment]
I recommend following this [video ](https://www.youtube.com/watch?v=ynR8RQ_DAJE&ab_channel=Elementor) with the hope that you find it beneficial.
